exchanges/stream/websocket.go

```diff
@@ -298,11 +296,11 @@ func (w *Websocket) dataMonitor() {
 }
 
 // connectionMonitor ensures that the WS keeps connecting
-func (w *Websocket) connectionMonitor() {
-	if w.IsConnectionMonitorRunning() {
-		return
+func (w *Websocket) connectionMonitor() (wasRunning bool) {
```

Comment:
I think this one should be an error if anything is to be returned. It brings up the question why is this being called when one is already running? Even if just to log the error to highlight that there is an error in our websocket process.